src/Data/Core.Caching/CacheManager.cs

```diff
@@ -1,30 +1,39 @@
-﻿using BeetleX.Redis;
-using Microsoft.Extensions.Logging;
+﻿using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace QuantumCore.Caching;
 
 public class CacheManager : ICacheManager
 {
     private readonly ILogger<CacheManager> _logger;
-    private readonly RedisDB _redis;
+
+    public IRedisStore Shared { get; }
+    public IRedisStore Server { get; }
+    
+    private IRedisStore _defaultRedisStore;
 
     public CacheManager(ILogger<CacheManager> logger, IOptions<CacheOptions> options)
     {
         _logger = logger;
-        _redis = new RedisDB { DataFormater = new JsonFormater() };
         _logger.LogInformation("Initialize Cache Manager");
-        _redis.Host.AddWriteHost(options.Value.Host, options.Value.Port);
+
+        Shared = new RedisStore(CacheStoreType.Shared, logger, options.Value);
```

Comment:
I'd like this to be injected instead of being constructed here. Maybe with a [keyed dependency](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-8.0#keyed-services)?

Reply:
Wow, great feedback, thank you! For the context, this is my first dotnet experience and sometimes I approach with my primitive insticts. These reviews helped me so much!

Reply:
You're welcome! C# is one thing, but knowing the default libraries by Microsoft and their latest features is a whole 'nother spiel